qa/tasks/cephadm.py

```diff
@@ -1133,3 +1135,51 @@ def task(ctx, config):
 
         finally:
             log.info('Teardown begin')
+
+
+def add_mirror(s, mirror: str):
+    config = toml.loads(s)
+    is_v1 = 'registries' in config
+    if is_v1:
+        search = config.get('registries', {}).get('search', {}).get('registries', [])
+        insecure = config.get('registries', {}).get('search', {}).get('insecure', [])
+        # v2: MutableMapping[str, Any] = { needs Python 3
+        v2 = {
+            'unqualified-search-registries': search,
+            'registry': [
+                {
+                    'prefix': reg,
+                    'location': reg,
+                    'insecure': reg in insecure,
+                    'blocked': False,
+                } for reg in search
+            ]
+        }
+    else:
+        v2 = config  # type: ignore
+    dockers = [r for r in v2['registry'] if r['prefix'] == 'docker.io']
+    if dockers:
+        docker = dockers[0]
+        docker['mirror'] = [{
+            "location": mirror,
+            "insecure": True,
+        }]
+    return v2
+
+
+def add_mirror_to_cluster(ctx, mirror: str):
```

Comment:
if you named it as `add_mirror_to_cluster` then signature should be `(mirror: str, cluster: teuthology.orchestra.cluster)`?
But it is better to give the name `add_containers_registry_mirror(mirror: str, cluster)` or something.